task/fbc-validation/0.1/fbc-validation.yaml

```diff
@@ -111,47 +134,84 @@ spec:
-        cp -r .$conffolder/* $(workspaces.workspace.path)/hacbs/$(context.task.name)/confdir
 
-        # We have totally 4 checks here currently
-        check_num=4
+        # copy content of conffolder to confdir - will be used in next task -
+        related image check
+
+        cp -r .$conffolder/*
+        $(workspaces.workspace.path)/hacbs/$(context.task.name)/confdir
+
+
+        EXTRACT_DIR="/extracted_base_img"
+
+        mkdir "${EXTRACT_DIR}"
+
+        oc image extract ${BASE_IMAGE} --path /:"${EXTRACT_DIR}"
+
+        OPM_BINARIES="$(find "${EXTRACT_DIR}" -type f -name opm)"
+
+        BINARIES_COUNT=$(wc -l <<< "${OPM_BINARIES}")
+
+        if [[ $BINARIES_COUNT -ne "1" ]]; then
+            echo "expected exactly one opm binary in base image; found $BINARIES_COUNT."
+            exit 0
+        fi
+
+        OPM_BINARY=$(echo "${OPM_BINARIES}" | head -n 1)
+
+        echo "OPM_BINARY: '${OPM_BINARY}'"
+
+        # We have totally 3 checks here currently
+
+        check_num=3
+
         failure_num=0
+
         TESTPASSED=true
 
-        if [[ ! $(find . -name "opm") ]]; then
-          echo "!FAILURE! - opm binary presence check failed."
-          failure_num=`expr $failure_num + 1`
-          TESTPASSED=false
-        fi
+
         if [[ ! $(find . -name "grpc_health_probe") ]]; then
           echo "!FAILURE! - grpc_health_probe binary presence check failed."
           failure_num=`expr $failure_num + 1`
           TESTPASSED=false
         fi
-        if ! opm validate ."${conffolder}"; then
+
+        if ! ${OPM_BINARY} validate ."${conffolder}"; then
           echo "!FAILURE! - opm validate check failed."
           failure_num=`expr $failure_num + 1`
           TESTPASSED=false
         fi
-        if ! opm render ."${conffolder}" | jq -en 'reduce (inputs | select(.schema == "olm.package")) as $obj (0; .+1) == 1'; then
+
+        if ! ${OPM_BINARY} render ."${conffolder}" | jq -en 'reduce (inputs |
+        select(.schema == "olm.package")) as $obj (0; .+1) == 1'; then
           echo "!FAILURE! - More than one olm.packages is not permitted in a FBC fragment."
           failure_num=`expr $failure_num + 1`
           TESTPASSED=false
         fi
-        note="Task $(context.task.name) completed: Check result for task result."
+
+        note="Task $(context.task.name) completed: Check result for task
+        result."
+
         if [ $TESTPASSED == false ]; then
           ERROR_OUTPUT=$(make_result_json -r FAILURE -f $failure_num -s `expr $check_num - $failure_num` -t "$note")
           echo "${ERROR_OUTPUT}" | tee $(results.TEST_OUTPUT.path)
         else
           TEST_OUTPUT=$(make_result_json -r SUCCESS -s $check_num -t "$note")
           echo "${TEST_OUTPUT}" | tee $(results.TEST_OUTPUT.path)
         fi
+
         popd
+      securityContext:
+        capabilities:
+          add:
+            - SETFCAP
+        runAsUser: 0
+      workingDir: $(workspaces.workspace.path)/hacbs/$(context.task.name)
```

Comment:
Why do we need PVC for working dir and just local dir? It makes operations slower.

Reply:
I think that was also stuff I accidentally copied back over from OpenShift when I was testing.  I'll remove it.